src/awkward/contents/indexedarray.py

```diff
@@ -1006,57 +1006,14 @@ def _reduce_next(
         if not branch and negaxis == depth:
             return out
         else:
-            if isinstance(out, ak.contents.RegularArray):
-                out = out.toListOffsetArray64(True)
-
             # If the result of `_reduce_next` is a list, and we're not applying at this
             # depth, then it will have offsets given by the boundaries in parents.
             # This means that we need to look at the _contents_ to which the `outindex`
             # belongs to add the new index
```

Comment:
This comment is for the code that was removed.

```suggestion
```

Reply:
This made me go back and stare at this function - we don't need much of this logic, because it's trivial in the case of non-option index types.